syntax = "proto3";

package yandex.cloud.audit.ydb;

import "google/protobuf/struct.proto";
import "google/protobuf/timestamp.proto";
import "google/rpc/status.proto";
import "yandex/cloud/audit/common.proto";
import "yandex/cloud/validation.proto";
import "yandex/cloud/ydb/v1/database.proto";
import "yandex/cloud/ydb/v1/backup.proto";

option go_package = "a.yandex-team.ru/cloud/bitbucket/public-api/yandex/cloud/audit/ydb;ydb";
option java_package = "yandex.cloud.api.audit.ydb";
option java_outer_classname = "YdbEvents";

message CreateDatabase {
  string event_id = 1 [(required) = true];
  string event_source = 2 [(required) = true];
  string event_type = 3 [(required) = true];
  google.protobuf.Timestamp event_time = 4 [(required) = true];
  Authentication authentication = 5 [(required) = true];
  Authorization authorization = 6 [(required) = true];
  ResourceMetadata resource_metadata = 7 [(required) = true];
  RequestMetadata request_metadata = 8 [(required) = true];
  EventStatus event_status = 9 [(required) = true];
  google.rpc.Status error = 10;
  EventDetails details = 11 [(required) = true];
  google.protobuf.Struct request_parameters = 12;
  google.protobuf.Struct response = 13;

  message EventDetails {
    string database_id = 1 [(required) = true];
    string database_name = 2 [(required) = true];
    oneof database_type {
      yandex.cloud.ydb.v1.DedicatedDatabase dedicated_database = 3;
      yandex.cloud.ydb.v1.ServerlessDatabase serverless_database = 4;
    }
    yandex.cloud.ydb.v1.Database.Status status = 5;
    string endpoint = 6;
    string resource_preset_id = 7;
    yandex.cloud.ydb.v1.StorageConfig storage_config = 8;
    yandex.cloud.ydb.v1.ScalePolicy scale_policy = 9;
    string network_id = 10;
    repeated string subnet_ids = 11;
    bool assign_public_ips = 12;
    string document_api_endpoint = 13;
    string kinesis_api_endpoint = 14;
    string location_id = 17 [(required) = true];
    yandex.cloud.ydb.v1.BackupConfig backup_config = 18;
  }
}

message StartDatabase {
  string event_id = 1 [(required) = true];
  string event_source = 2 [(required) = true];
  string event_type = 3 [(required) = true];
  google.protobuf.Timestamp event_time = 4 [(required) = true];
  Authentication authentication = 5 [(required) = true];
  Authorization authorization = 6 [(required) = true];
  ResourceMetadata resource_metadata = 7 [(required) = true];
  RequestMetadata request_metadata = 8 [(required) = true];
  EventStatus event_status = 9 [(required) = true];
  google.rpc.Status error = 10;
  EventDetails details = 11 [(required) = true];
  google.protobuf.Struct request_parameters = 12;
  google.protobuf.Struct response = 13;

  message EventDetails {
    string database_id = 1 [(required) = true];
    string database_name = 2 [(required) = true];
    oneof database_type {
      yandex.cloud.ydb.v1.DedicatedDatabase dedicated_database = 3;
      yandex.cloud.ydb.v1.ServerlessDatabase serverless_database = 4;
    }
    string location_id = 5 [(required) = true];
    string endpoint = 6;
    string resource_preset_id = 7;
    yandex.cloud.ydb.v1.StorageConfig storage_config = 8;
    yandex.cloud.ydb.v1.ScalePolicy scale_policy = 9;
    string network_id = 10;
    repeated string subnet_ids = 11;
    bool assign_public_ips = 12;
    string document_api_endpoint = 13;
    string kinesis_api_endpoint = 14;
    yandex.cloud.ydb.v1.Database.Status status = 15;
    yandex.cloud.ydb.v1.BackupConfig backup_config = 18;
  }
}

message StopDatabase {
  string event_id = 1 [(required) = true];
  string event_source = 2 [(required) = true];
  string event_type = 3 [(required) = true];
  google.protobuf.Timestamp event_time = 4 [(required) = true];
  Authentication authentication = 5 [(required) = true];
  Authorization authorization = 6 [(required) = true];
  ResourceMetadata resource_metadata = 7 [(required) = true];
  RequestMetadata request_metadata = 8 [(required) = true];
  EventStatus event_status = 9 [(required) = true];
  google.rpc.Status error = 10;
  EventDetails details = 11 [(required) = true];
  google.protobuf.Struct request_parameters = 12;
  google.protobuf.Struct response = 13;

  message EventDetails {
    string database_id = 1 [(required) = true];
    string database_name = 2 [(required) = true];
    oneof database_type {
      yandex.cloud.ydb.v1.DedicatedDatabase dedicated_database = 3;
      yandex.cloud.ydb.v1.ServerlessDatabase serverless_database = 4;
    }
    string location_id = 5 [(required) = true];
    string endpoint = 6;
    string resource_preset_id = 7;
    yandex.cloud.ydb.v1.StorageConfig storage_config = 8;
    yandex.cloud.ydb.v1.ScalePolicy scale_policy = 9;
    string network_id = 10;
    repeated string subnet_ids = 11;
    bool assign_public_ips = 12;
    string document_api_endpoint = 13;
    string kinesis_api_endpoint = 14;
    yandex.cloud.ydb.v1.Database.Status status = 15;
    yandex.cloud.ydb.v1.BackupConfig backup_config = 18;
  }
}

message UpdateDatabase {
  string event_id = 1 [(required) = true];
  string event_source = 2 [(required) = true];
  string event_type = 3 [(required) = true];
  google.protobuf.Timestamp event_time = 4 [(required) = true];
  Authentication authentication = 5 [(required) = true];
  Authorization authorization = 6 [(required) = true];
  ResourceMetadata resource_metadata = 7 [(required) = true];
  RequestMetadata request_metadata = 8 [(required) = true];
  EventStatus event_status = 9 [(required) = true];
  google.rpc.Status error = 10;
  EventDetails details = 11 [(required) = true];
  google.protobuf.Struct request_parameters = 12;
  google.protobuf.Struct response = 13;

  message EventDetails {
    string database_id = 1 [(required) = true];
    string database_name = 2 [(required) = true];
    oneof database_type {
      yandex.cloud.ydb.v1.DedicatedDatabase dedicated_database = 3;
      yandex.cloud.ydb.v1.ServerlessDatabase serverless_database = 4;
    }
    string location_id = 5 [(required) = true];
    string endpoint = 6;
    string resource_preset_id = 7;
    yandex.cloud.ydb.v1.StorageConfig storage_config = 8;
    yandex.cloud.ydb.v1.ScalePolicy scale_policy = 9;
    string network_id = 10;
    repeated string subnet_ids = 11;
    bool assign_public_ips = 12;
    string document_api_endpoint = 13;
    string kinesis_api_endpoint = 14;
    yandex.cloud.ydb.v1.Database.Status status = 15;
    yandex.cloud.ydb.v1.BackupConfig backup_config = 18;
  }
}

message DeleteDatabase {
  string event_id = 1 [(required) = true];
  string event_source = 2 [(required) = true];
  string event_type = 3 [(required) = true];
  google.protobuf.Timestamp event_time = 4 [(required) = true];
  Authentication authentication = 5 [(required) = true];
  Authorization authorization = 6 [(required) = true];
  ResourceMetadata resource_metadata = 7 [(required) = true];
  RequestMetadata request_metadata = 8 [(required) = true];
  EventStatus event_status = 9 [(required) = true];
  google.rpc.Status error = 10;
  EventDetails details = 11 [(required) = true];
  google.protobuf.Struct request_parameters = 12;
  google.protobuf.Struct response = 13;

  message EventDetails {
    string database_id = 1 [(required) = true];
    string database_name = 2 [(required) = true];
    oneof database_type {
      yandex.cloud.ydb.v1.DedicatedDatabase dedicated_database = 3;
      yandex.cloud.ydb.v1.ServerlessDatabase serverless_database = 4;
    }
    string location_id = 5 [(required) = true];
    string endpoint = 6;
    string resource_preset_id = 7;
    yandex.cloud.ydb.v1.StorageConfig storage_config = 8;
    yandex.cloud.ydb.v1.ScalePolicy scale_policy = 9;
    string network_id = 10;
    repeated string subnet_ids = 11;
    bool assign_public_ips = 12;
    string document_api_endpoint = 13;
    string kinesis_api_endpoint = 14;
    yandex.cloud.ydb.v1.Database.Status status = 15;
    yandex.cloud.ydb.v1.BackupConfig backup_config = 18;
  }
}

message BackupDatabase {
  string event_id = 1 [(required) = true];
  string event_source = 2 [(required) = true];
  string event_type = 3 [(required) = true];
  google.protobuf.Timestamp event_time = 4 [(required) = true];
  Authentication authentication = 5 [(required) = true];
  Authorization authorization = 6 [(required) = true];
  ResourceMetadata resource_metadata = 7 [(required) = true];
  RequestMetadata request_metadata = 8 [(required) = true];
  EventStatus event_status = 9 [(required) = true];
  google.rpc.Status error = 10;
  EventDetails details = 11 [(required) = true];
  google.protobuf.Struct request_parameters = 12;
  google.protobuf.Struct response = 13;

  message EventDetails {
    string database_id = 1 [(required) = true];
    string database_name = 2 [(required) = true];
    string backup_id = 3 [(required) = true];
    google.protobuf.Timestamp created_at = 4 [(required) = true];
    yandex.cloud.ydb.v1.BackupSettings backup_settings = 5;
    string backup_name = 6;
    google.protobuf.Timestamp started_at = 7;
    google.protobuf.Timestamp completed_at = 8;
    yandex.cloud.ydb.v1.Backup.Status status = 9;
    yandex.cloud.ydb.v1.Backup.Type type = 10;
    int64 size = 11;
  }
}

message RestoreBackup {
  string event_id = 1 [(required) = true];
  string event_source = 2 [(required) = true];
  string event_type = 3 [(required) = true];
  google.protobuf.Timestamp event_time = 4 [(required) = true];
  Authentication authentication = 5 [(required) = true];
  Authorization authorization = 6 [(required) = true];
  ResourceMetadata resource_metadata = 7 [(required) = true];
  RequestMetadata request_metadata = 8 [(required) = true];
  EventStatus event_status = 9 [(required) = true];
  google.rpc.Status error = 10;
  EventDetails details = 11 [(required) = true];
  google.protobuf.Struct request_parameters = 12;
  google.protobuf.Struct response = 13;

  message EventDetails {
    string database_id = 1 [(required) = true];
    string database_name = 2 [(required) = true];
    string backup_id = 3 [(required) = true];
    string target_path = 4 [(required) = true];
    repeated string paths_to_restore = 5;
    string backup_name = 6;
    google.protobuf.Timestamp started_at = 7;
    google.protobuf.Timestamp completed_at = 8;
    yandex.cloud.ydb.v1.Backup.Status status = 9;
    yandex.cloud.ydb.v1.Backup.Type type = 10;
    int64 size = 11;
  }
}

message DeleteBackup {
  string event_id = 1 [(required) = true];
  string event_source = 2 [(required) = true];
  string event_type = 3 [(required) = true];
  google.protobuf.Timestamp event_time = 4 [(required) = true];
  Authentication authentication = 5 [(required) = true];
  Authorization authorization = 6 [(required) = true];
  ResourceMetadata resource_metadata = 7 [(required) = true];
  RequestMetadata request_metadata = 8 [(required) = true];
  EventStatus event_status = 9 [(required) = true];
  google.rpc.Status error = 10;
  EventDetails details = 11 [(required) = true];
  google.protobuf.Struct request_parameters = 12;
  google.protobuf.Struct response = 13;

  message EventDetails {
    string database_id = 1 [(required) = true];
    string database_name = 2 [(required) = true];
    string backup_id = 3 [(required) = true];
    google.protobuf.Timestamp created_at = 4 [(required) = true];
    string backup_name = 5;
    google.protobuf.Timestamp started_at = 6;
    google.protobuf.Timestamp completed_at = 7;
    yandex.cloud.ydb.v1.Backup.Status status = 8;
    yandex.cloud.ydb.v1.Backup.Type type = 9;
    int64 size = 10;
  }
}
